### ISSUE TITLE
Add test warning output to JUnit XML file

### ISSUE THREAD

**Issue by timj** [pytest-dev/pytest#2717]
The new feature where pytest captures all the warnings and reports them after it runs is great. Unfortunately, those warnings do not appear in the JUnit XML file and that means that our CI system does not report the warnings to the user. Is it possible to capture the warnings from each and store them in the appropriate stderr section in the XML file? It seems like warnings are intercepted before they can be caught by the JUnit code. Making them visible to our CI users would motivate them to fix them.

Thanks again for pytest.

**Comment by nicoddemus** [comment]
Hi @timj thanks for writing.

In which section of the XML report do you feel that would be appropriate? After a quick glance at the [junit schema](https://github.com/windyroad/JUnit-Schema/blob/master/JUnit.xsd) I didn't see anywhere  obvious...

**Comment by timj** [comment]
I think for warnings I would expect it to be written to the same place that standard error ends up: `<xs:element name="system-err">`.

**Comment by timj** [comment]
Yes, so I think your accumulating of warnings is stopping them from going to standard error and therefore stopping them from going in to the JUnit file.

**Comment by kjeldflarup** [comment]
Wouldn't it be more obvious to generate a 

> <xs:element name="warning">

 in the same place as 

> <xs:element name="failure">

**Comment by RonnyPfannschmidt** [comment]
as per the junitxml file format, warnings are not supported/express-able but we could output them as another system-err node as a hack

**Comment by RonnyPfannschmidt** [comment]
hmm, actuall i noted that would be invalid

**Comment by RonnyPfannschmidt** [comment]
hmm - based on the relaxng shema of junitmxl i just stumbled uppon pytest does **everyything** wrong with junitxml ^^

**Comment by timj** [comment]
I'm not overly hung up at this point on where the warnings go in the JUnit XML. My concern is that they disappear completely so people using the nice Jenkins test report UI don't see them. I'm fine with them going in with all the other standard output in the XML.

**Comment by kanetkarster** [comment]
Sorry for bumping an old issue, but I think adding 
```
filterwarnings = error
```
to the pytest conf would make most users happy with the junit output for warnings

**Comment by timj** [comment]
My reading of the docs is that that would mean all my tests start failing that generate warnings? I don't want to force everyone to stop what they are doing and fix the warnings. I want the warnings to be visible so that people can see them and plan a timeline for fixing them. A FutureWarning is useful for planning but should not break things.

**Comment by piotr-dobrogost** [comment]
1. I found out about warnings' capture in pytest
2. I added them to code under tests
3. I saw Jenkins report does not contain them
4. I found this issue

(Writing all of above just to avoid simple but hated _bump_)

**Comment by RonnyPfannschmidt** [comment]
After a quick Google search on the evolution and current capabilities of junitxml, my impression is that it ought to be moved out of the core pytest framework and handed over to maintainers that care about junitxml
